### PR TITLE
#119 Add default presentation

### PR DESCRIPTION
**Pull request by dennis95stumm**
# Fixes \#119

## Type (Highlight the corresponding type)
- Bugfix
- **Feature**
- Documentation
- Refactoring (e.g. Style updates, Test implementation, etc.)
- Other (please describe):

## Checklist
- [x] Code updated to current masters head
- [x] Passes CI checks
- [x] Is a part of an issue
- [x] Tests added for the bugfix or newly implemented feature, describe below why if not
- [x] Changelog is updated
- [x] Documentation of code and features exists

## Changes

- Added a configurable system wide default presentation

### CODE REVIEW

File: app/Http/Controllers/api/v1/ApplicationController.php [hunk 1]
```diff
@@ -75,4 +87,24 @@ public function currentUser()
...
         return (new UserResource(Auth::user()))->withPermissions();
     }
+
+    /**
+     * Returns the default presentation or 404 if there is no default presentation
+     * or the file is missing.
+     *
+     * @return StreamedResponse
+     */
+    public function defaultPresentation(): StreamedResponse
```
Comment (SamuelWei) [inline app/Http/Controllers/api/v1/ApplicationController.php:97]: Could we somehow use this https://laravel.com/docs/8.x/filesystem#the-public-disk so we don't have to stream the default_presentation via php and can just use the webserver.

Only for the private presentations we have to use the Storage::download to ensure the access is restricted.

**Review by SamuelWei — CHANGES_REQUESTED** [review]

File: resources/js/views/settings/Application.vue [hunk 2]
```diff
@@ -166,139 +166,212 @@
...
+                id='default_presentation'
+                :state="fieldState('default_presentation')"
+                :browse-text="$t('app.browse')"
+                :placeholder="$t('settings.application.defaultPresentation')"
+                v-model="default_presentation"
+              >
+              </b-form-file>
+              <b-form-text v-if="!viewOnly">{{ $t('rooms.files.formats', { formats: settings.bbb.file_mimes }) }}<br>{{ $t('rooms.files.size', { size: settings.bbb.max_filesize }) }}</b-form-text>
+              <template slot='invalid-feedback' v-if="!viewOnly">
+                <div v-html="fieldError('default_presentation')"></div>
```
Comment (SamuelWei) [inline resources/js/views/settings/Application.vue:370]: missing translation of the form validator

File: resources/js/views/settings/Application.vue [hunk 3]
```diff
@@ -166,139 +166,212 @@
...
+            </b-form-group>
+          </b-col>
+        </b-row>
+
+        <b-row cols='12'>
+          <b-col md='6'>
+            <b-form-group
+              label-class="font-weight-bold"
+              class="mb-4"
+              :label="$t('settings.application.default_timezone')"
```
Comment (SamuelWei) [inline resources/js/views/settings/Application.vue:289]: wrong naming convention. We don't have it equal through the whole app, but let's try to at least use the same standard in a component/view.

File: resources/js/views/settings/Application.vue [hunk 4]
```diff
@@ -166,139 +166,212 @@
...
-          >
-          </b-form-input>
-
-          <template slot='invalid-feedback'>
-            <div v-html="fieldError('pagination_page_size')"></div>
-          </template>
-        </b-form-group>
+        <b-row cols='12'>
+          <b-col md='6'>
+            <b-form-group
```
Comment (SamuelWei) [inline resources/js/views/settings/Application.vue:171]: I would prefer a heading like all the other settings 'block's:
image

**Review by SamuelWei — CHANGES_REQUESTED** [review]

File: app/Meeting.php [hunk 5]
```diff
@@ -102,6 +102,10 @@ public function startMeeting()
             $meetingParams->addPresentation($file->getDownloadLink(), null, preg_replace("/[^A-Za-z0-9.-_\(\)]/", '', $file->filename));
         }
 
+        if (empty($meetingParams->getPresentations()) && !empty(setting('default_presentation'))) {
+            $meetingParams->addPresentation(setting('default_presentation'));
```
Comment (SamuelWei) [inline app/Meeting.php:106]: I would prefer a presentation name, so BBB does not use this random name. We could hardcode "default.pdf", use a translated string or a custom variable. @dennis95stumm What do you think?

Comment (dennis95stumm) [reply app/Meeting.php:106]: I've hardcoded the name with the extension from the client!

Comment (SamuelWei) [reply app/Meeting.php:106]: That's a great solution, I did not even think about the extension

**Pull request merged**